src/Sulu/Bundle/EventLogBundle/Application/Collector/DomainEventCollector.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\EventLogBundle\Application\Collector;
+
+use Doctrine\Common\EventSubscriber;
+use Doctrine\ORM\Event\OnClearEventArgs;
+use Doctrine\ORM\Event\PostFlushEventArgs;
+use Doctrine\ORM\Events;
+use Sulu\Bundle\EventLogBundle\Application\Dispatcher\DomainEventDispatcherInterface;
+use Sulu\Bundle\EventLogBundle\Domain\Event\DomainEvent;
+
+class DomainEventCollector implements DomainEventCollectorInterface, EventSubscriber
+{
+    /**
+     * @var DomainEventDispatcherInterface
+     */
+    private $domainEventDispatcher;
+
+    /**
+     * @var DomainEvent[]
+     */
+    private $eventsToBeDispatched = [];
+
+    public function __construct(
+        DomainEventDispatcherInterface $domainEventDispatcher
+    ) {
+        $this->domainEventDispatcher = $domainEventDispatcher;
+    }
+
+    public function getSubscribedEvents()
+    {
+        return [
+            Events::onClear,
+            Events::postFlush,
+        ];
+    }
+
+    public function collect(DomainEvent $domainEvent): void
+    {
+        $this->eventsToBeDispatched[] = $domainEvent;
+    }
+
+    public function onClear(OnClearEventArgs $args): void
+    {
+        $this->eventsToBeDispatched = [];
+    }
+
+    public function postFlush(PostFlushEventArgs $args): void
```

Comment:
yes but that would make the interface bigger and we would need to do something similar in the `DocumentDomainEventCollector`. not sure if it is worth that because i do not really see any advantage of exposing the methods

Reply:
The advantage is possible to call them in an import command yourself and it would be cleaner as the Collector does not depend on the infrastructure events. Currently `postFlush` and `onClear` would also need to be part of the `DomainEventCollector` if somebody would try to decorate the service.

Reply:
done 🙂